### DISCUSSION TITLE
repos.yaml and addons.yaml doesnt work for me

### DISCUSSION THREAD

**Discussion by mork02** [Tecnativa/doodba#617] [General]
I'm trying to import the "https://github.com/OCA/sale-workflow/tree/16.0/sale_force_invoiced" repository and somehow everything t tried didnt work. 
This is my repos.yaml:

./odoo:
  defaults:
    # Shallow repositories ($DEPTH_DEFAULT=1) are faster & thinner
    # You may need a bigger depth when merging PRs (use $DEPTH_MERGE
    # for a sane value of 100 commits)
    depth: $DEPTH_DEFAULT
  remotes:
    ocb: https://github.com/OCA/OCB.git
    odoo: https://github.com/odoo/odoo.git
    openupgrade: https://github.com/OCA/OpenUpgrade.git
  target: ocb $ODOO_VERSION
  merges:
    - ocb $ODOO_VERSION
    # Example of a merge of the PR with the number <PR>
    # - oca refs/pull/<PR>/head

./sale-workflow:
  defaults:
    depth: $DEPTH_DEFAULT
  remotes:
    sale_workflow: https://github.com/OCA/sale-workflow.git
  target: sale_workflow 15.0
  merges:
    - sale_workflow 15.0
    - 
and this is my addons.yaml:

./sale-workflow:
  - sale_force_invoiced

**Comment by mork02** [comment]
Im using docker-compose images

**Reply by pedrobaeza** [reply]
No need to add nothing extra in repos.yaml, as OCA repositories are already handled automatically.

Just add to addons.yaml

```
sale-workflow:
  - sale_force_invoiced
```

No need to put `./`.

Then, it depends if you are in devel or prod/demo mode. For devel, do `invoke git-aggregate` restart Odoo, and update module list to find it. For prod/demo, build the image, up -d and update module list.

**Reply by mork02** [reply]
i get this error:" failed to solve: failed to compute cache key: failed to calculate checksum of ref 3195c0ca-8989-4ab9-8588-810b06c967f7::e8g338yng4yvy4dk0bwd1vgzr: "/qa": not found"
Do i just need to mkdir the file somewhere?

**Reply by pedrobaeza** [reply]
If you want a help with quality, you need to provide information with the same good quality. You get that error doing what? In which environment? Whole steps to reproduce the problem?

**Reply by mork02** [reply]
I use odoo 15 with psql 13 Images. I run everytthing on my local maschiene with Docker-compose and Docker Desktop. When i use **invoke git-aggregate** i get this error.

**Reply by josep-tecnativa** [reply]
Could you show us your full repos.yaml and addons.yaml and also the exact commands you are using please?

**Reply by mork02** [reply]
repos.yaml

`./odoo:
  defaults:
    # Shallow repositories ($DEPTH_DEFAULT=1) are faster & thinner
    # You may need a bigger depth when merging PRs (use $DEPTH_MERGE
    # for a sane value of 100 commits)
    depth: $DEPTH_DEFAULT
  remotes:
    ocb: https://github.com/OCA/OCB.git
    odoo: https://github.com/odoo/odoo.git
    openupgrade: https://github.com/OCA/OpenUpgrade.git
  target: ocb $ODOO_VERSION
  merges:
    - ocb $ODOO_VERSION
    # Example of a merge of the PR with the number <PR>
    # - oca refs/pull/<PR>/head

./sale-workflow:
  defaults:
    depth: $DEPTH_DEFAULT
  remotes:
    sale_workflow: https://github.com/OCA/sale-workflow.git
  target: main 15.0
  merges:
    - main 15.0`
    
    
    addons.yaml
    `sale-workflow:
  - sale_force_invoiced`
  
  
  docker-compose.yml
  `version: '3.8'

services:
  odoo:
    image: odoo:15.0
    env_file: .env
    depends_on:
      - postgres
    ports:
      - "127.0.0.1:8069:8069"
    volumes:
      - data:/var/lib/odoo
      - ./custom/src/private:/mnt/extra-addons
    environment:
      - HOST=postgres
      - PORT=5432
      - USER=odoo
      - PASSWORD=123

  postgres:
    image: postgres:13
    env_file: .env
    environment:
      POSTGRES_USER: odoo
      POSTGRES_PASSWORD: 123
      POSTGRES_DB: postgres
    volumes:
      - db:/var/lib/postgresql/data

volumes:
  data:
  db:
`

**Reply by pedrobaeza** [reply]
That error seems something of Docker itself. Are you using it in Windows? If so, it seems there's something incorrect in the installation.

**Reply by pedrobaeza** [reply]
And have you populated the scaffolding with the copier template in https://github.com/Tecnativa/doodba-copier-template/?

**Reply by mork02** [reply]
im using Linux mint. So do i need to deinstall my whole docker images, valumes?

**Reply by mork02** [reply]
Yes i installed Doodba copier template

**Reply by pedrobaeza** [reply]
But the `docker-compose.yml` doesn't correspond to Doodba Copier Template. You have to make a symbolic link of docker-compose.yaml to devel.yaml, prod.yaml or test.yaml.

**Reply by mork02** [reply]
How to do that?

**Reply by josep-tecnativa** [reply]
When you say you installed Doodba copier template, Do you mean that you followed the DCT readme steps?
In particular, I mean that to generate the scaffolding of a project, you should use the command indicated in this part of the README:
https://github.com/Tecnativa/doodba-copier-template?tab=readme-ov-file#use-the-template-to-generate-your-subproject

**Reply by josep-tecnativa** [reply]
It is also important to install dependencies before.
https://github.com/Tecnativa/doodba-copier-template?tab=readme-ov-file#install-the-dependencies

**Reply by pedrobaeza** [reply]
For example:

`ln -s devel.yaml docker-compose.yml`

but if you remove the current file and launch `invoke develop`, it's done automatically.

**Reply by mork02** [reply]
found all of my problems. Thank you!

**Reply by pedrobaeza** [reply]
Well, it's interesting for others with the same problem that you put here your discoverings.

**Reply by mork02** [reply]
My biggest problem was that i didnt installed doodba right and i broke my linux users rights. 
Installing Doodba copier: 
1. Copy copier template
2. run invoke git-aggrigate
3. run docker-compose build
4. run docker-compose run odoo odoo -i base
5. run docker-compose up
for clearly exploration follow the doodba readme

**Reply by mork02** [reply]
and in my case when i  use oca modules it only must be declared in the addons.yaml not in the repos.yaml
